### ISSUE TITLE
Shapes layer: 'below' doesn't play well with subplots

### ISSUE THREAD

**Issue by alexcjohnson** [plotly/plotly.js#1452]
I'm not sure this really *has* a solution even theoretically, if you consider the layering that needs to happen with inset plots (where the background of the inset plot needs to be drawn on top of the data of the main plot) but posting as a bug for discussion purposes since it's at the very least unexpected.

If you make a `layer: 'below'` shape that's data-referenced on one axis and paper-referenced on another, and it crosses two subplots (which could actually be a very nice way to highlight correlations between these subplots) the shape will NOT be below the data of both subplots.

```js
Plotly.newPlot(gd,[
    {y:[1,2,3]},
    {y:[1,2,3],yaxis:'y2'}
],{
    yaxis:{domain:[0,0.4]},
    yaxis2:{domain:[0.6,1]},
    shapes:[{
        xref:'x',yref:'paper',type:'rect',
        x0:0.8,x1:1.2,y0:0,y1:1,
        fillcolor:'#eee',layer:'below'
    }]
})
```

![screen shot 2017-03-08 at 12 38 35 am](https://cloud.githubusercontent.com/assets/2678795/23691624/9e86b9f2-0397-11e7-846f-dfe8d0b87d3f.png)

**Comment by etpinard** [comment]
Nice find. I hope the [`shape_below_traces`](https://github.com/plotly/plotly.js/blob/master/test/image/mocks/shapes_below_traces.json) isn't [wrong](https://github.com/plotly/plotly.js/blob/master/test/image/baselines/shapes_below_traces.png).

I'd vote for making `layer: 'below'` correspongd to below **all** traces linked to the axis set in the shape `xref` or `yref`.

**Comment by alexcjohnson** [comment]
closed by #1453 - now `layer: 'below'` shapes and images are still in front of plot backgrounds in all cases EXCEPT on inset plots (any subplot that overlaps an earlier one without being explicitly overlaying) where the component is NOT referenced to both axes of that subplot. So for example, if `x2/y2` is on top of `x/y`, a shape referenced to `x2/y2` can still be seen with `layer: below` but one referenced to `x2/paper`, `paper/y2`, `x/y`, or anything else cannot be seen where it intersects the `x2/y2` subplot.

This is, I believe, the minimal set of objects it's theoretically impossible to show (short of some sort of horribly complicated stitching together of different objects in different z layers, which wouldn't work that well anyway due to antialiasing and would have strange effects of data objects from the lower plot slicing through them.)
